From e1a8d5e66efd0e492716e67e0db4cdedc723e741 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Tue, 14 May 2013 10:41:06 -0500
Subject: [PATCH 152/210] Change method of detection for daemonize

When changing the source from kernel 3.8- to 3.9+, a stale version of
include/linux/version.h might be left in the source, which causes the
test for the availability of daemonize to fail. This patch makes the
test be more robust.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 include/osdep_service.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rtl8192du/include/osdep_service.h
+++ b/drivers/staging/rtl8192du/include/osdep_service.h
@@ -365,7 +365,7 @@ static inline unsigned char _cancel_time
 
 static __inline void thread_enter(char *name)
 {
-	#if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 8, 0))
+	#ifdef daemonize
 	daemonize("%s", name);
 	#endif
 	allow_signal(SIGTERM);
